src/pages/index.js

```diff
@@ -0,0 +1,236 @@
+/* API */
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "bba6e655-3c43-4bc9-84b1-5706718b60cd",
+    "Content-Type": "application/json",
+  },
+});
+
+let section;
+
+api;
+
+Promise.all([api.getInitialCards(), api.getUserInfo()]).then(
+  ([cards, data]) => {
+    section = new Section(
+      {
+        items: cards,
+        renderer: (data) => {
+          const cardEl = renderCard(data);
+          section.addItem(cardEl);
+        },
+      },
+      ".gallery__cards"
+    );
+    section.renderItems();
+
+    userInfo.setUserInfo({
+      title: data.name,
+      description: data.about,
+    });
+
+    userInfo.setAvatar({ avatar: data.avatar });
+  }
+);
+
+const userInfo = new UserInfo({
+  titleSelector: ".profile__title",
+  descriptionSelector: ".profile__description",
+  avatarSelector: ".profile__image",
+});
+
+function handleImageClick({ name, link }) {
+  popupWithImage.open(name, link);
+}
+
+function renderCard(data) {
+  const card = new Card(
+    data,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCard,
+    handleLike
+  );
+  console.log(card);
+  return card.getView();
+}
+
+function handleProfileEditSubmit({ title, description }) {
+  editModal.setLoading(true);
+  api
+    .setUserInfo(title, description)
+    .then(() => {
+      userInfo.setUserInfo({ name: title, about: description });
+      editModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editModal.setLoading(false);
+    });
+}
+
+function handleAddCardSubmit(name, url) {
+  addModal.setLoading(true);
+  api
+    .addCard(name, url)
+    .then((data) => {
+      const card = renderCard(data);
+      section.addItem(card);
+      addModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      addModal.setLoading(false);
+    });
+}
+
+function handleAvatarSubmit(url) {
+  profileAvatarPopUp.setLoading(true);
+  api
+    .updateAvatar(url)
+    .then((data) => {
+      userInfo.getAvatar(data);
+      profileAvatarPopUp.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileAvatarPopUp.setLoading(false);
+    });
+}
+
+function handleDeleteCard(card) {
+  cardDeletePopUp.open();
+  cardDeletePopUp.setSubmitAction(() => {
+    cardDeletePopUp.setLoading(true, "Deleting...");
+    api
+      .deleteCard(card.id)
+      .then(() => {
+        card.handleDeleteCard();
+        cardDeletePopUp.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        cardDeletePopUp.setLoading(false, "Yes");
+      });
+  });
+} //fixed the delete!!!
+
+function handleLike(cardInstance) {
+  if (cardInstance.isLiked) {
+    api
+      .dislikeCard(cardInstance.id)
+      .then(() => {
+        cardInstance.handleLikeIcon();
+        cardInstance.isLiked = false;
```

Comment:
Needs correction: please don't change the likes state directly, use `updateIsLiked` method for updating the likes state and rendering the button